### PR TITLE
feat/KFS-1058-exit attached mode upon pressing ControlC, ControlD, ControlQ..

### PR DESCRIPTION
**Pull request by Guttz**
Release Notes
-------------

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

When submitting persistent queries like INSERT INTO that run in the background we attach to the job and listen for errors and the user can exit this mode now pressing also ctrlC, ctrlD and so on, which are the regular cancelEvents for the other statements. 
 
References
----------

Test & Review
-------------

unit tests
Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/pkg/flink/internal/controller/statement_controller.go [hunk 1]
```diff
@@ -125,14 +113,33 @@ func (c *StatementController) waitForStatementToBeInTerminalStateOrError(process
...
+		return false
+	}
+}
+
+func isCancelEvent(key prompt.Key) bool {
+	switch key {
+	case prompt.ControlC, prompt.ControlD, prompt.ControlQ, prompt.Escape:
+		return true
+	}
+	return false
```
Comment (brianstrauch) [inline internal/pkg/flink/internal/controller/statement_controller.go:136]:
```suggestion
	return types.Contains([]string{prompt.ControlC, prompt.ControlD, prompt.ControlQ, prompt.Escape}, key)
```

File: internal/pkg/flink/internal/controller/statement_controller_test.go [hunk 2]
```diff
@@ -315,3 +315,75 @@ func (s *StatementControllerTestSuite) TestRenderMsgAndStatusNonLocalNonFailedSt
...
+			want: true,
+		},
+		{
+			name: "Other",
+			key:  prompt.ShiftDown, // Just an example of a key that is not in the switch cases
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (brianstrauch) [inline internal/pkg/flink/internal/controller/statement_controller_test.go:352]:
```suggestion
	for _, test := range tests {
```

File: internal/pkg/flink/internal/controller/statement_controller.go (hunk 1 above)
Comment (YannickPferr) [reply internal/pkg/flink/internal/controller/statement_controller.go:136]: Neat one-liner! You could also do it with lo like this:
`lo.Contains([]prompt.Key{prompt.ControlC, prompt.ControlD, prompt.ControlQ, prompt.Escape}, pressedKey)`

File: internal/pkg/flink/internal/controller/statement_controller.go [hunk 3]
```diff
@@ -125,14 +113,33 @@ func (c *StatementController) waitForStatementToBeInTerminalStateOrError(process
...
+}
+
+func isDetachEvent(key prompt.Key) bool {
+	switch key {
```
Comment (YannickPferr) [inline internal/pkg/flink/internal/controller/statement_controller.go:140]: and then you could also reduce this to one line:
`lo.Contains([]prompt.Key{prompt.ControlM, prompt.Enter}, key)`

**Review by YannickPferr — APPROVED** [review]
Thanks!

**Pull request merged**